x-pack/plugins/fleet/server/services/epm/package_service.ts

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { KibanaRequest, ElasticsearchClient, SavedObjectsClient, Logger } from 'kibana/server';
+import type {
+  EsAssetReference,
+  InstallablePackage,
+  RegistryPackage,
+  RegistrySearchResult,
+} from '../../types';
+import { ElasticsearchAssetType } from '../../types';
+import { checkSuperuser } from '../../routes/security';
+import { FleetUnauthorizedError } from '../../errors';
+import { getPathParts } from './archive';
+import { installTransform } from './elasticsearch/transform/install';
+import { fetchFindLatestPackage, getRegistryPackage } from './registry';
+
+export type InstalledAssetType = EsAssetReference;
```

Comment:
not all of the installs return `EsAssetReference` so this is just prepping for eventual need to add more return types.